sklearn/decomposition/tests/test_sparse_pca.py

```diff
@@ -168,32 +178,34 @@ def test_mini_batch_fit_transform():
     assert_array_almost_equal(spca_lasso.components_, spca_lars.components_)
 
 
-def test_scaling_fit_transform():
+def test_scaling_fit_transform(global_random_seed):
     alpha = 1
-    rng = np.random.RandomState(0)
+    rng = np.random.RandomState(global_random_seed)
     Y, _, _ = generate_toy_data(3, 1000, (8, 8), random_state=rng)
     spca_lars = SparsePCA(n_components=3, method="lars", alpha=alpha, random_state=rng)
     results_train = spca_lars.fit_transform(Y)
     results_test = spca_lars.transform(Y[:10])
     assert_allclose(results_train[0], results_test[0])
 
 
-def test_pca_vs_spca():
-    rng = np.random.RandomState(0)
+def test_pca_vs_spca(global_random_seed):
+    rng = np.random.RandomState(global_random_seed)
     Y, _, _ = generate_toy_data(3, 1000, (8, 8), random_state=rng)
     Z, _, _ = generate_toy_data(3, 10, (8, 8), random_state=rng)
-    spca = SparsePCA(alpha=0, ridge_alpha=0, n_components=2)
-    pca = PCA(n_components=2)
+    spca = SparsePCA(
+        alpha=0, ridge_alpha=0, n_components=2, random_state=global_random_seed
+    )
+    pca = PCA(n_components=2, random_state=global_random_seed)
```

Comment:
It's not required to start from the same seedin the test, is it ?
```suggestion
    spca = SparsePCA(
        alpha=0, ridge_alpha=0, n_components=2, random_state=rng
    )
    pca = PCA(n_components=2, random_state=rng)
```

Reply:
Thank you. 
I need to ask (maybe @glemaitre ?) why sometimes it's ok to use `rng` (from np.random.RandomState(global_random_seed)`)..  and sometimes the same seed.

Reply:
`global_random_seed` is a number like 0, 1, 42, ...
`np.random.RandomState` is an object with a state. When you ask it to generate a random value its state changes, so the next time you ast it again it gives a different value.
If you pass an integer as random state to an estimator, it's used to create and seed a RandomState object. Therefore, each time it's fitted, it starts from the same seed and produces the same results.

So to answer your question, it depends if need to generate multiple times the exact same sequence of generated value or if it doesn't matter.
For instance, if you want to compare the results of an estimator between float32 and float64 input, you want that both estimators use the same sequence of generated values.

Reply:
Thank you very much @jeremiedbb... I see that my question wasn't clear.

In the file modified with this PR, some tests set `SparcePCA` parameter `random_state` as `random_state=rng`, where `rng = np.random.RandomState(global_random_seed)`.  For example `test_fit_transform_tall`. 

But some tests set the `random_state` parameter as `random_state=global_random_seed`. Like `test_sparse_pca_numerical_consistency`. 

I guess it doesn't matter as there is this `check_random_state(seed)` that  "Turn seed into a np.random.RandomState instance". 

In the end they might both be the same, except that changing random_state of `test_sparse_pca_numerical_consistency` to `random_state=rng` was failing. 

Anyway, your previous explanation confirmed my understanding. :-)